src/measure_theory/measurable_space.lean

```diff
@@ -134,15 +135,67 @@ lemma monotone_comap : monotone (measurable_space.comap g) := assume a b h, coma
 lemma comap_map_le : (m.map f).comap f ≤ m := (gc_comap_map f).l_u_le _
 lemma le_map_comap : m ≤ (m.comap g).map g := (gc_comap_map g).le_u_l _
 
+lemma comap_of_involutive {m' : measurable_space β} {g : β → β} (hg : involutive g)
+  (hg' : measurable g) (f : α → β) :
+  measurable_space.comap (λ a, g (f a)) infer_instance = measurable_space.comap f infer_instance :=
+begin
+  ext,
+  set e : set β ≃ set β :=
+  { to_fun := preimage g,
+    inv_fun := preimage g,
+    left_inv := hg.preimage,
+    right_inv := hg.preimage },
+  refine e.exists_congr_left.trans (exists_congr $ λ t, _),
+  simp only [preimage_preimage, compl_compl, equiv.coe_fn_symm_mk, and.congr_left_iff, hg _],
+  rintro rfl,
+  refine ⟨λ ht, _, λ ht, hg' ht⟩,
+  convert hg' ht,
+  simp_rw [preimage_preimage, hg _, preimage_id'],
+end
+
+lemma comap_compl {m' : measurable_space β} [boolean_algebra β] (h : measurable (compl : β → β))
+  (f : α → β) :
+  measurable_space.comap (λ a, (f a)ᶜ) infer_instance = measurable_space.comap f infer_instance :=
+comap_of_involutive compl_involutive h _
+
 end functors
 
+@[simp] lemma map_const {m} (b : β) : measurable_space.map (λ a : α, b) m = ⊤ :=
+eq_top_iff.2 $ by { rintro s hs, by_cases b ∈ s; change measurable_set (preimage _ _); simp [*] }
+
+@[simp] lemma comap_const {m} (b : β) : measurable_space.comap (λ a : α, b) m = ⊥ :=
+eq_bot_iff.2 $ by { rintro _ ⟨s, -, rfl⟩, by_cases b ∈ s; simp [*] }
+
 lemma comap_generate_from {f : α → β} {s : set (set β)} :
   (generate_from s).comap f = generate_from (preimage f '' s) :=
 le_antisymm
   (comap_le_iff_le_map.2 $ generate_from_le $ assume t hts,
     generate_measurable.basic _ $ mem_image_of_mem _ $ hts)
   (generate_from_le $ assume t ⟨u, hu, eq⟩, eq ▸ ⟨u, generate_measurable.basic _ hu, rfl⟩)
 
+/-- The sigma-algebra generated by a single set `s` is `{∅, s, sᶜ, univ}`. -/
+@[simp] lemma generate_from_singleton (s : set α) :
+  generate_from {s} = measurable_space.comap (∈ s) ⊤ :=
+begin
+  classical,
+  refine le_antisymm (generate_from_le $ λ t ht, ⟨{true}, trivial, by simp [ht.symm]⟩) _,
```

Comment:
Do we have lemmas `measurable (p : α → Prop) ↔ measurable_set {x | p x}` and `measurable (∈ s)  ↔ measurable_set s`? Probably, not. Could you add it and use here, please?

Reply:
We definitely don't since before this PR we don't have a `measurable_space Prop` instance.